source/algorithms.tex

```diff
@@ -3540,17 +3540,18 @@
   for the overloads with both parameters \tcode{pred} and \tcode{proj1}.
 \end{itemize}
 
+\pnum
+Let $N$ be $\min(\tcode{last1 - first1}, \ \tcode{last2 - first2})$.
+
 \pnum
 \returns
 \tcode{\{ first1 + n, first2 + n \}},
-where \tcode{n} is the smallest integer such that $E$ holds,
-or $\min(\tcode{last1 - first1}, \ \tcode{last2 - first2})$
-if no such integer exists.
+where \tcode{n} is the smallest integer less than $N$ such that $E$ holds,
```

Comment:
... smallest integer in `[0, N)` ...

Otherwise what stops this from picking a negative value of `n`?

Reply:
Fixed.